### PR TITLE
Improved Home Page #44

### PR DESCRIPTION
**Pull request by nishant0708**
Improved Home Page #44

- Utilized GSAP library and ScrollTrigger for enhanced animations on our homepage.
- Images fade in from 0 to 1 opacity smoothly on page load.
- Hero text elegantly slides in from left to right during initial load, ensuring responsiveness across devices.
- Applied reverse animation to hero-text-2, with text entering from the right on scroll trigger, maintaining responsiveness.
- Synchronized image reveal animation with text effects for a cohesive visual experience on all screen sizes.
- Added subtle up-and-down motion to images, enhancing visual appeal and responsiveness.
- These improvements contribute to a dynamic and engaging user experience across various devices.
- Excited to contribute these responsive enhancements to our open-source project.

### CODE REVIEW

**Comment by afeefuddin** [comment]
The animation in which the text is appearing seems unnecessary and doesn't looks for a professional website. We don't have any plans to implement it. You should drop that part of the PR at least. The Image animation is Okayish.

**Comment by nishant0708** [comment]
Okay I will remove text animation and raising pr

**Comment by afeefuddin** [comment]
Don't raise a new PR, work on this one and modify commits and rebase them accordingly.

**Comment by afeefuddin** [comment]
There are 4 commits, there should be 1 commit.

**Comment by nishant0708** [comment]
okay

**Comment by nishant0708** [comment]
I have squashed all my commits into one, please check.

File: frontend/src/app/page.tsx [hunk 1]
```diff
@@ -4,15 +4,61 @@ import hero_img from "../assets/hero_img.png";
...
+import { gsap } from "gsap";
+import { ScrollTrigger } from "gsap/ScrollTrigger";
 
 export default function Home() {
   const router = useRouter();
 
+  gsap.registerPlugin(ScrollTrigger);
+
+
+
```
Comment (afeefuddin) [inline frontend/src/app/page.tsx:17]: Remove 2 extra lines from here

File: frontend/src/app/page.tsx [hunk 2]
```diff
@@ -4,15 +4,61 @@ import hero_img from "../assets/hero_img.png";
...
+          trigger: ".hero_text1",
+          start: "top 40%",
+          end: "bottom top",
+        },
+      }
+    );
+  }, []);
+
+
+
```
Comment (afeefuddin) [inline frontend/src/app/page.tsx:37]: Remove 2 extra lines from here also.

**Comment by nishant0708** [comment]
I have removed extra lines  you asked for. Please check..

**Comment by afeefuddin** [comment]
LGTM. Merged @nishant0708, Thanks.

**Pull request merged**